src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -91,16 +109,48 @@ export default class ZipPaymentStrategy implements PaymentStrategy {
                             },
                         });
                     })
-                    .then(nonce =>
-                        this._store.dispatch(this._paymentActionCreator.submitPayment({
-                            methodId: payment.methodId,
-                            paymentData: { nonce },
-                        }))
+                    .then(nonce => {
+                            if (isNonceLike(nonce)) {
+                                return this._store.dispatch(this._paymentActionCreator.submitPayment({
+                                    methodId: payment.methodId,
+                                    paymentData: nonce,
+                                }));
+                            } else {
+                                this._goToOrderConfirmation();
+
+                                return this._store.getState();
+                            }
+                        }
                     ));
             });
     }
 
     finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
+
+    private _prepareForReferredRegistration(checkoutId: string): Promise<Response> {
+        const url = `checkout.php?action=zip_referred_registration`;
```

Comment:
Don't use `` if you are not doing any string interpolation.

Reply:
Also, what type of endpoint is that? It's not a versioned storefront or internal API endpoint. And it's not a legacy endpoint (i.e.: `set_external_checkout`). Do you just want to post some arbitrary data to the server? I believe there's a better way to achieve it (i.e.: via provider-specific endpoints).